src/main/java/hudson/remoting/RemoteInvocationHandler.java

```diff
@@ -935,12 +971,47 @@ private Method choose(Class[] interfaces) {
             return arguments;
         }
 
+        @Override
         public String toString() {
             return "RPCRequest("+oid+","+methodName+")";
         }
 
         private static final long serialVersionUID = 1L; 
     }
 
+    /**
+     * User-space version of {@link RPCRequest}.
+     *
+     * This is an equivalent of {@link UserRequest} for RPC calls.
+     * Such kind of requests will not be send over closing or malfunctional channel.
+     *
+     * If used as {@link Callable} in conjunction with {@link UserRequest},
+     * this can be used to send a method call to user-level objects, and
+     * classes for the parameters and the return value are sent remotely if needed.
+     */
+    static class UserRPCRequest extends RPCRequest {
+        public UserRPCRequest(int oid, Method m, Object[] arguments, ClassLoader cl) {
```

Comment:
I think you forgot to actually use this! Line 292.

Reply:
oh my :( I will make sure tests actually catch it

Reply:
done